BASE/src/Microsoft.ApplicationInsights/Extensibility/Implementation/Tracing/DiagnosticsModule/FileDiagnosticsSender.cs

```diff
@@ -0,0 +1,161 @@
+﻿namespace Microsoft.ApplicationInsights.Extensibility.Implementation.Tracing.DiagnosticsModule
+{
+    using System;
+    using System.Diagnostics;
+    using System.IO;
+
+    using Microsoft.ApplicationInsights.Common.Extensions;
+
+    using static System.FormattableString;
+
+    /// <summary>
+    /// This sender works with the DiagnosticTelemetryModule. This will subscribe to events and output to a text file log.
+    /// </summary>
+    internal class FileDiagnosticsSender : IDiagnosticsSender, IDisposable
+    {
+        private readonly object lockObj = new object();
+        private readonly string logFileName = FileHelper.GenerateFileName();
+        private readonly DefaultTraceListener defaultTraceListener = new DefaultTraceListener();
+
+        private string logDirectory = Environment.ExpandEnvironmentVariables("%TEMP%");
+        private bool isEnabled = false; // TODO: NEED MORE PERFORMANT FILE WRITTER BEFORE ENABLING THIS BY DEFAULT
```

Comment:
turning this on/off and what the default value should be  - i think this should be the resposibility of DiagnosticmModule.